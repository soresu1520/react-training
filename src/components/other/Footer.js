import styled from "styled-components";

const Footer = () => {
  return (
    <FooterDiv>
      <ElementDiv>
        <FooterText>Terms of Service</FooterText>
        <FooterText>Security</FooterText>
        <FooterText>Privacy</FooterText>
        <FooterText>Contact</FooterText>
        <FooterText>About us</FooterText>
      </ElementDiv>

      <ElementDiv>
        <Image src="/assets/socialmedia/twitter-fill.png" alt="twitter" />
        <Image src="/assets/socialmedia/instagram-line.png" alt="instagram" />
        <Image src="/assets/socialmedia/facebook-fill.png" alt="facebook" />
      </ElementDiv>

      <FooterCopyright>
        {"@ "}
        {new Date().getFullYear()} Click and Eat
      </FooterCopyright>
    </FooterDiv>
  );
};

export default Footer;

const FooterDiv = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  align-content: center;
`;

const ElementDiv = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
`;

const Image = styled.img`
  padding-left: 1rem;
  padding-right: 1rem;
`;

const FooterCopyright = styled.h5`
  font-size: var(--font-size-h5);
  color: var(--color-greyscale-200);
  font-weight: var(--font-weight-heading-light);
  margin-top: 1em;
`;

const FooterText = styled.h5`
  font-size: var(--font-size-h5);
  color: var(--color-greyscale-600);
  font-weight: var(--font-weight-heading-light);
  padding-left: 1rem;
  padding-right: 1rem;
  margin-bottom: 1em;
`;
