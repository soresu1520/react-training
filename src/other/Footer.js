import styled from "styled-components";
import facebook from "./assets/facebook-fill.png";
import twitter from "./assets/twitter-fill.png";
import instagram from "./assets/instagram-line.png";

const Footer = () => {
  return (
    <FooterDiv>
      <ElementDiv>
        <FooterText>Terms of Service</FooterText>
        <FooterText>Security</FooterText>
        <FooterText>Privacy</FooterText>
        <FooterText>Contact</FooterText>
        <FooterText>About us</FooterText>
      </ElementDiv>

      <ElementDiv>
        <Image src={twitter} alt="twitter" />
        <Image src={instagram} alt="instagram" />
        <Image src={facebook} alt="facebook" />
      </ElementDiv>

      <FooterCopyright>
        {"@ "}
        {new Date().getFullYear()} Click and Eat
      </FooterCopyright>
    </FooterDiv>
  );
};

export default Footer;

const FooterDiv = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  align-content: center;
`;

const ElementDiv = styled.div`
  display: flex;
  flex-direction: row;
  flex-flow: row wrap;
  justify-content: center;
`;

const Image = styled.img`
  padding-left: 1rem;
  padding-right: 1rem;
`;

const FooterCopyright = styled.h5`
  font-size: 0.85rem;
  color: var(--color-greyscale-200);
  font-weight: 400;
  margin-top: 1em;
`;

const FooterText = styled.h5`
  font-size: 0.85rem;
  color: var(--color-greyscale-600);
  font-weight: 400;
  padding-left: 1rem;
  padding-right: 1rem;
  margin-bottom: 1em;
`;
