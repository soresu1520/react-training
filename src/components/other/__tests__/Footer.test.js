import renderer from "react-test-renderer";
import Footer from "../Footer";

it("footer renders correctly", () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
