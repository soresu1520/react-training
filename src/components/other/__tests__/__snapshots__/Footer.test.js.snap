// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer renders correctly 1`] = `
<div
  className="sc-bczRLJ eQrVgT"
>
  <div
    className="sc-gsnTZi MdYWD"
  >
    <h5
      className="sc-eCYdqJ cWVpU"
    >
      Terms of Service
    </h5>
    <h5
      className="sc-eCYdqJ cWVpU"
    >
      Security
    </h5>
    <h5
      className="sc-eCYdqJ cWVpU"
    >
      Privacy
    </h5>
    <h5
      className="sc-eCYdqJ cWVpU"
    >
      Contact
    </h5>
    <h5
      className="sc-eCYdqJ cWVpU"
    >
      About us
    </h5>
  </div>
  <div
    className="sc-gsnTZi MdYWD"
  >
    <img
      alt="twitter"
      className="sc-dkzDqf kbKXbi"
      src="/assets/socialmedia/twitter-fill.png"
    />
    <img
      alt="instagram"
      className="sc-dkzDqf kbKXbi"
      src="/assets/socialmedia/instagram-line.png"
    />
    <img
      alt="facebook"
      className="sc-dkzDqf kbKXbi"
      src="/assets/socialmedia/facebook-fill.png"
    />
  </div>
  <h5
    className="sc-hKMtZM iQfqIu"
  >
    @ 
    2022
     Click and Eat
  </h5>
</div>
`;
